data-prepper-api/src/main/java/org/opensearch/dataprepper/model/configuration/DataPrepperVersion.java

```diff
@@ -0,0 +1,101 @@
+    private static final String CURRENT_VERSION = "2.1";
+
+    private static final String FULL_FORMAT = "%d.%d";
+    private static final String SHORTHAND_FORMAT = "%d";
+    private static final String VERSION_PATTERN_STRING = "^((\\d+)(\\.(\\d+))?)$";
+    private static final Pattern VERSION_PATTERN = Pattern.compile(VERSION_PATTERN_STRING);
+    private static final int MAJOR_VERSION_PATTERN_POSITION = 2;
+    private static final int MINOR_VERSION_PATTERN_POSITION = 4;
+
+    private final int majorVersion;
+    private final Integer minorVersion;
+
+    private static DataPrepperVersion instance;
+
+    private DataPrepperVersion(int majorVersion, Integer minorVersion) {
+        this.minorVersion = minorVersion;
+        this.majorVersion = majorVersion;
+    }
+
+    public static DataPrepperVersion getCurrentVersion() {
+        if (Objects.isNull(instance)) {
+            instance = parse(CURRENT_VERSION);
+        }
+        return instance;
+    }
+
+    public static DataPrepperVersion parse(final String version) {
+
+        final Matcher result = VERSION_PATTERN.matcher(version);
+        if(result.find()) {
+            String major = result.group(MAJOR_VERSION_PATTERN_POSITION);
+            final int foundMajorVersion = Integer.parseInt(major);
+            final String potentialMinorVersion = result.group(MINOR_VERSION_PATTERN_POSITION);
+            final Integer foundMinorVersion = potentialMinorVersion == null ? null : Integer.parseInt(potentialMinorVersion);
+
+            return new DataPrepperVersion(foundMajorVersion, foundMinorVersion);
+        }
+
+        throw new IllegalArgumentException("Invalid Data Prepper Version provided: " + version);
+    }
+
+    public int getMajorVersion() {
+        return majorVersion;
+    }
+
+    public Optional<Integer> getMinorVersion() {
+        return Optional.ofNullable(minorVersion);
+    }
+
+    /**
+     * Determines if the provided Data Prepper Version is compatible with this.
+     *
+     * The initial implementation is a basic implementation that enforces equivalent versions, any shorthand format version
+     * compared with full format with equivalent major versions and equivalent major version but the comparing minor version
+     * is less than this minor version are compatible.
+     *
+     * @param o - the other DataPrepperVersion to compare with
+     * @return return true if the versions are compatible, otherwise false
+     * @since 2.1
+     */
+    public boolean compatibleWith(DataPrepperVersion o) {
+
+        if (this.majorVersion != o.getMajorVersion()) {
+            return false;
+        }
+
+        if (this.minorVersion != null && o.getMinorVersion().isPresent()) {
+            if (!this.minorVersion.equals(o.getMinorVersion().get())) {
+                return this.minorVersion > o.getMinorVersion().get();
+            }
+        }
+
+        return true;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
We should also have an implementation for hashCode.

Reply:
Can do!